src/Microsoft.Health.Fhir.SecretStore/Registration/FhirServerBuilderSecretStoreRegistrationExtensions.cs

```diff
@@ -0,0 +1,53 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using Microsoft.Azure.KeyVault;
+using Microsoft.Azure.Services.AppAuthentication;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Health.Extensions.DependencyInjection;
+using Microsoft.Health.Fhir.Core.Features.SecretStore;
+using Microsoft.Health.Fhir.Core.Registration;
+using Microsoft.Health.Fhir.KeyVault;
+using Microsoft.Health.Fhir.KeyVault.Configs;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class FhirServerBuilderSecretStoreRegistrationExtensions
+    {
+        private const string KeyVaultConfigurationName = "KeyVault";
+
+        public static IFhirServerBuilder AddSecretStore(this IFhirServerBuilder fhirServerBuilder, IConfiguration configuration)
+        {
+            EnsureArg.IsNotNull(fhirServerBuilder, nameof(fhirServerBuilder));
+            EnsureArg.IsNotNull(configuration, nameof(configuration));
+
+            var keyVaultConfig = new KeyVaultConfiguration();
+            configuration.GetSection(KeyVaultConfigurationName).Bind(keyVaultConfig);
```

Comment:
KeyVault section might not exit at all (e.g., if the KeyVault section is not in the appsettings.json file).

Reply:
I looked at the `GetSection` call and it will return an empty IConfigurationSection in case it can't find the specified sub-section in the config file. So I guess we should be fine since we want to anyway create an InMemorySecretStore in such cases.